src/huggingface_hub/lfs.py

```diff
@@ -389,20 +390,37 @@ def _upload_parts_hf_transfer(
             " not available in your environment. Try `pip install hf_transfer`."
         )
 
-    try:
-        return multipart_upload(
-            file_path=operation.path_or_fileobj,
-            parts_urls=sorted_parts_urls,
-            chunk_size=chunk_size,
-            max_files=128,
-            parallel_failures=127,  # could be removed
-            max_retries=5,
+    supports_callback = "callback" in inspect.signature(multipart_upload).parameters
+    if not supports_callback:
+        warnings.warn(
+            "You are using an outdated version of `hf_transfer`. Consider upgrading to latest version to enable progress bars using `pip install -U hf_transfer`."
         )
-    except Exception as e:
-        raise RuntimeError(
-            "An error occurred while uploading using `hf_transfer`. Consider disabling HF_HUB_ENABLE_HF_TRANSFER for"
-            " better error handling."
-        ) from e
+
+    total = operation.upload_info.size
+    desc = operation.path_in_repo
+    if len(desc) > 40:
+        desc = f"(…){desc[-40:]}"
+    disable = bool(logger.getEffectiveLevel() == logging.NOTSET)
+
+    with tqdm(unit="B", unit_scale=True, total=total, initial=0, desc=desc, disable=disable) as progress:
+        try:
+            output = multipart_upload(
+                file_path=operation.path_or_fileobj,
+                parts_urls=sorted_parts_urls,
+                chunk_size=chunk_size,
+                max_files=128,
+                parallel_failures=127,  # could be removed
+                max_retries=5,
+                **({"callback": progress.update} if supports_callback else {}),
+            )
+            if not supports_callback:
+                progress.update(total)
```

Comment:
Maybe that this could be done outside of the try block ?

Reply:
Oh yes, good catch!